<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>867</width>
    <height>650</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>867</width>
    <height>650</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>867</width>
    <height>650</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>3D Viewer</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="glView" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>510</width>
      <height>580</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>350</y>
      <width>141</width>
      <height>100</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="pushButton_3">
       <property name="text">
        <string>vertices color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>background color</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>edges color</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>10</y>
      <width>301</width>
      <height>329</height>
     </rect>
    </property>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>translate by ox</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="ox"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>translate by oy</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="oy"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>translate by oz</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="oz"/>
     </item>
     <item row="3" column="1">
      <widget class="QPushButton" name="startTranslate">
       <property name="text">
        <string>translate</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>rotate by ox</string>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QLineEdit" name="rot_ox"/>
     </item>
     <item row="5" column="0">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>rotate by oy</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QLineEdit" name="rot_oy"/>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>rotate by oz</string>
       </property>
      </widget>
     </item>
     <item row="6" column="1">
      <widget class="QLineEdit" name="rot_oz"/>
     </item>
     <item row="7" column="1">
      <widget class="QPushButton" name="startRotate">
       <property name="text">
        <string>rotate</string>
       </property>
      </widget>
     </item>
     <item row="8" column="1">
      <widget class="QLineEdit" name="scale"/>
     </item>
     <item row="9" column="1">
      <widget class="QPushButton" name="scale_model">
       <property name="text">
        <string>scale</string>
       </property>
      </widget>
     </item>
     <item row="8" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>scale the model</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>530</y>
      <width>141</width>
      <height>47</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>Vertice size</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="vertice_size"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>590</y>
      <width>511</width>
      <height>18</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="file_name">
       <property name="text">
        <string>File name:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="vertices">
       <property name="text">
        <string>Vertices count:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="edges">
       <property name="text">
        <string>Edges count:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>460</y>
      <width>114</width>
      <height>63</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Choose lines type:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="smooth_button">
       <property name="text">
        <string>smooth</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="stipple_button">
       <property name="text">
        <string>stipple</string>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
       <property name="autoRepeat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>460</y>
      <width>141</width>
      <height>47</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Line thickness</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="thickness"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>530</y>
      <width>111</width>
      <height>81</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_4">
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Vertices type:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="vertice_none">
       <property name="text">
        <string>none</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="vertice_circle">
       <property name="text">
        <string>circle</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="vertice_square">
       <property name="text">
        <string>square</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>350</y>
      <width>111</width>
      <height>62</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <item>
      <widget class="QLabel" name="label_13">
       <property name="text">
        <string>Projection:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="orthogonal_button">
       <property name="text">
        <string>Orthogonal</string>
       </property>
       <property name="checkable">
        <bool>true</bool>
       </property>
       <property name="checked">
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="perspective_button">
       <property name="text">
        <string>Perspective</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>867</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuopen_file">
    <property name="title">
     <string>Open File</string>
    </property>
    <addaction name="actionopen"/>
   </widget>
   <widget class="QMenu" name="menuGif">
    <property name="title">
     <string>Gif</string>
    </property>
    <addaction name="actionSave_as_Gif"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionSave_Image"/>
   </widget>
   <addaction name="menuopen_file"/>
   <addaction name="menuGif"/>
   <addaction name="menuImage"/>
  </widget>
  <action name="actionopen">
   <property name="text">
    <string>open..</string>
   </property>
  </action>
  <action name="actionSave_settings">
   <property name="text">
    <string>Save settings</string>
   </property>
  </action>
  <action name="actionSave_as_Gif">
   <property name="text">
    <string>Save Gif</string>
   </property>
  </action>
  <action name="actionSave_Image">
   <property name="text">
    <string>Save Image</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>glView</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">myopenglwidjet.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
